python/pyarrow/table.pxi

```diff
@@ -1638,26 +1638,45 @@ def table(data, names=None, schema=None, metadata=None):
             "Expected pandas DataFrame, python dictionary or list of arrays")
 
 
-def concat_tables(tables):
+def concat_tables(tables, with_promotion=False, MemoryPool memory_pool=None):
     """
-    Perform zero-copy concatenation of pyarrow.Table objects. Raises exception
-    if all of the Table schemas are not the same
+    Perform zero-copy concatenation of pyarrow.Table objects.
+    if with_promotion==True, Columns of the same name will be concatenated.
+    They should be of the same type, or be of type NULL, in which case it will
+    be promoted to the type of other corresponding columns with null values
+    filled.  If a table is missing a particular field, null values of the
+    appropriate type will be generated to take the place of the missing field.
+    The new schema will share the metadata with the first table. Each field in
+    the new schema will share the metadata with the first table which has the
+    field defined.
+    if with_promotion==False, the schemas of all the Tables must be the same,
```

Comment:
```suggestion
    Ff with_promotion==False (the default), the schemas of all the Tables must be the same,
```

Can you put this first? (before the explanation of `with_promotion=True`) Since this is the default behaviour, seems more logical to start with it.

Also a question: if we say that the "schemas of all Tables must be the same", does that also mean that the metadata need to be the same?